data-prepper-api/src/main/java/org/opensearch/dataprepper/model/source/SourceCoordinationStore.java

```diff
@@ -33,7 +34,7 @@ boolean tryCreatePartitionItem(final String partitionKey,
      * 2. The partition status is CLOSED and the reOpenAt timestamp has passed
      * 3. The partition status is ASSIGNED and the partitionOwnershipTimeout has passed
      */
-    Optional<SourcePartitionStoreItem> tryAcquireAvailablePartition(final String ownerId, final Duration ownershipTimeout);
+    Optional<SourcePartitionStoreItem> tryAcquireAvailablePartition(final String sourceIdentifier, final String ownerId, final Duration ownershipTimeout);
```

Comment:
I am seeing the following javadocs warnings here (when I compile in my workspace). You should fix it in your next PR.
```
/local/home/krishkdk/data-prepper-arith-exp2/kk-data-prepper-f2/data-prepper-api/src/main/java/org/opensearch/dataprepper/model/source/SourceCoordinationStore.java:36: warning: no @param for ownerId
    Optional<SourcePartitionStoreItem> tryAcquireAvailablePartition(final String ownerId, final Duration ownershipTimeout);
                                       ^
/local/home/krishkdk/data-prepper-arith-exp2/kk-data-prepper-f2/data-prepper-api/src/main/java/org/opensearch/dataprepper/model/source/SourceCoordinationStore.java:36: warning: no @param for ownershipTimeout
    Optional<SourcePartitionStoreItem> tryAcquireAvailablePartition(final String ownerId, final Duration ownershipTimeout);
                                       ^
/local/home/krishkdk/data-prepper-arith-exp2/kk-data-prepper-f2/data-prepper-api/src/main/java/org/opensearch/dataprepper/model/source/SourceCoordinationStore.java:36: warning: no @return
    Optional<SourcePartitionStoreItem> tryAcquireAvailablePartition(final String ownerId, final Duration ownershipTimeout);
                                       ^
3 warnings

```

Reply:
I'll fix in the next PR